src/main/java/io/aiven/kafka/auth/json/AivenAcl.java

```diff
@@ -38,26 +38,34 @@ public class AivenAcl {
     @SerializedName("resource_pattern")
     public final String resourceRePattern;
 
+    @SerializedName("permission_type")
+    private final AclPermissionType permissionType;
```

Comment:
There's a nasty problem with this enum: if you type `deny` instead of `DENY`, it'll be silently interpreted as `ALLOW`. That's not great from the user experience point of view and welcomes bad configuration bugs, especially considering it's a security feature. Could we either accept any casing or fail as invalid if we read `deny` or anything else?

Reply:
I added a custom deserializer for `AclPermissionType`. It is now case insensitive, it deals explicitly with null values and it throws an exception if the permission type is not recognized.